src/com/github/Kraken3/AFKPGC/BotDetector.java

```diff
@@ -0,0 +1,227 @@
+package com.github.Kraken3.AFKPGC;
+
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.Date;
+import java.util.LinkedList;
+import java.util.Map;
+import java.util.Set;
+import java.util.UUID;
+
+import org.bukkit.BanList;
+import org.bukkit.Bukkit;
+import org.bukkit.Location;
+import org.bukkit.entity.Player;
+
+
+/*
+ * @author Maxopoly
+ */
+public class BotDetector implements Runnable {
+	public static boolean longBans;
+	public static float currentTPS = 20;
+	public static float acceptableTPS;
+	public static float criticalTPSChange;
+	public static long frequency; // how often this runs in ticks
+	public static File banfile;
+	float lastRoundTPS;
+	UUID topSuspect = null;
+	String topSuspectName = ""; // needs to be stored additionally,
+								// because retrieving the name off
+								// players after kicking them is only
+								// possible
+								// over deprecated/buggy methods
+	Location topSuspectsLocation = null;
+	UUID lastRoundSuspect = null;
+	String lastRoundSuspectName = "";
+	Location lastRoundSuspectsLocation = null;
+	public static LinkedList<String> suspectedBotters = new LinkedList<String>();
+	public static LinkedList<String> bannedPlayers = new LinkedList<String>();
+	// this is needed as a separated list, so we know the difference between
+	// players who were banned by AFKPGC and players who were banned for other
+	// reasons
+
+	static BanList banList = AFKPGC.plugin.getServer().getBanList(
+			BanList.Type.NAME); // ban after names not ips
+
+	public void run() {
+		if (!AFKPGC.enabled) {
+			return;
+		}
+		currentTPS = TpsReader.getTPS();
+		Map<UUID, LastActivity> lastActivities = LastActivity.lastActivities;
+		if (currentTPS < acceptableTPS) {
+			int smallestMovedDistance = 1024;
+			Set<Map.Entry<UUID, LastActivity>> entries = lastActivities
+					.entrySet();
+			for (Map.Entry<UUID, LastActivity> entry : entries) {
+				UUID playerUUID = entry.getKey();
+				if (lastActivities.containsKey(playerUUID)) { // according to
+																// the author of
+																// AFKGPC, there

```

Comment:
Edit your line width settings to auto-fit at 120 chars; giant whitespace voids like this b/c of end of line comments are pretty terrible. Alternatively, just move the comment as the first line of the ```if``` block.